internal/templates/render/render.go

```diff
@@ -0,0 +1,177 @@
+	forceOverwrite  bool
+	keepTempDirs    bool
+	inputs          map[string]string
+}
+
+// Desc implements cli.Command.
+func (c *Command) Desc() string {
+	return "Instantiate a template to setup a new app or add config files"
+}
+
+func (c *Command) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The {{ COMMAND }} command renders the given template into the given dest
+  directory.
+`
+}
+
+func (c *Command) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	sect := set.NewSection("Render options")
+	sect.StringVar(&cli.StringVar{
+		Name:    "template",
+		Aliases: []string{"t"},
+		Example: "helloworld@v1",
+		Target:  &c.template,
+		Usage: `the location of the template to be instantiated. Many forms are accepted. ` +
+			`"helloworld@v1" means "github.com/abcxyz/helloworld repo at revision v1; this is for a template owned by abcxyz. ` +
+			`"myorg/helloworld@v1" means github.com/myorg/helloworld repo at revision v1; this is for a template not owned by abcxyz but still on GitHub. ` +
+			`"mygithost.com/mygitrepo/helloworld@v1" is for a template in a remote git repo but not owned by abcxyz and not on GitHub. ` +
+			`"mylocaltarball.tgz" is for a template not in git but present on the local filesystem. ` +
+			`"http://example.com/myremotetarball.tgz" os for a non-Git template in a remote tarball.`,
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "spec",
+		Aliases: []string{"s"},
+		Example: "path/to/spec.yaml",
+		Target:  &c.spec,
+		Default: "./spec.yaml",
+		Usage:   "the path of the .yaml file within the template directory that specifies how the template is rendered.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "dest",
+		Aliases: []string{"d"},
+		Example: "/my/git/dir",
+		Target:  &c.dest,
+		Default: ".",
+		Usage:   "the target directory in which to write the output files.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "git-protocol",
+		Aliases: []string{"p"},
+		Example: "https",
+		Target:  &c.gitProtocol,
+		Usage:   "either ssh or https, the protocol for connecting to GitHub.",
+	})
+	sect.StringMapVar(&cli.StringMapVar{
+		Name:    "input",
+		Aliases: []string{"i"},
+		Example: "foo=bar",
+		Target:  &c.inputs,
+		Usage:   "key=val pairs of template values; may be repeated.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "log-level",
+		Aliases: []string{"l"},
+		Example: "info",
+		Default: "warning",
+		Target:  &c.logLevel,
+		Usage:   "how verbose to log; any of debug|info|warning|error.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "allow-non-git-dest",
+		Aliases: []string{"a"},
+		Target:  &c.allowNonGitDest,
+		Usage:   "skip the sanity checks that the dest directory is a git repo.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "force-overwrite",
+		Aliases: []string{"o"},
+		Target:  &c.forceOverwrite,
+		Usage:   "if an output file already exists in the destination, overwrite it instead of failing.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "keep-temp-dirs-on-failure",
```

Comment:
Do we need to make it special for failures?
```suggestion
		Name:    "keep-temp-dirs",
```

Reply:
Good point, fixed.